tests/mssqltestutils.py

```diff
@@ -90,29 +92,78 @@ def getTempPath(*args):
     return  os.path.abspath(tempPath)
 
 def create_test_db():
-    client = create_mssql_cli_client()
+    """
+    Creates database for test, using various status checks and retry logic for reliability.
+    - Calls helper method to check status of create db task, if possible
+    - Exits on successful response or retry period exceeds time limit
+    """
+    options = create_mssql_cli_options(database='master')
+    client = create_mssql_cli_client(options)
+
+    if not _is_client_db_on_cloud(client):
```

Comment:
We should be able to run on-premise. Rather than raising an exception, can you just skip the rety logic for on-premise?